devito/finite_difference.py

```diff
@@ -190,3 +191,34 @@ def first_derivative(*args, **kwargs):
             var = [a.subs({dim: ind[i]}) for a in args]
             deriv += c[i] * reduce(mul, var, 1)
     return matvec._transpose*deriv
+
+
+def generic_derivative(function, order, dim, width):
+    """
+    Create generic arbitrary order derivative expression from a
+    single :class:`Function` object. This methods is essentially a
+    dedicated wrapper around SymPy's `as_finite_diff` utility for
+    :class:`devito.Function` objects.
+
+    :param function: The symbol representing a function.
+    :param order: Derivative order, eg. 2 for a second derivative.
+    :param dim: The dimension for which to take the derivative.
+    :param width: Order of the coefficient discretization and thus
```

Comment:
Nitpicking,

 I know it says what it does, but `width` is an odd name, would call it fd_order or something like that

Reply:
Agreed, will change.